src/big_core_base.cpp

```diff
@@ -1,120 +1,99 @@
-// RV: general comment. Vsechny komentare, ktere pravdepodobne nebudou soucasti finalni verze a jsou to spise otazky nebo odpovedi na me komentare prosim oznacuj inicialy LB, podobne jako to delam ja. Takoveto komentare mohou byt cesky bez diakritiky.
-// RV: general comment. V cpp souborech dodrzuj prosim pokud mozno stejne poradi metod jako v hlavickovem souboru. Lepe se pak hleda a kontroluje, co je naimplementovano.
-
-BigCoreBase::BigCoreBase()
-    // RV: nasledujici dva radky bych smazal, nevim proc tam cokoli nastavovat, dokud nemam parametry (rozmery, datove typy, atd.)
-	//:
-	//outermostEntitiesOffsets(0)
-{}
-
 BigCoreBase::~BigCoreBase()
 {
-    if (this->data) delete[] this->data;
-}
-
-uint64_t BigCoreBase::getNumberOfImages()
-{
-	return this->numberOfImages;
-}
-
-uint64_t BigCoreBase::getNumberOfTiles()
-{
-	return this->numberOfTiles;
-}
-
-uint64_t BigCoreBase::getImageHeight()
-{
-	return this->imageHeight;
-}
-
-uint64_t BigCoreBase::getImageWidth()
-{
-	return this->imageWidth;
-}
-
-uint64_t BigCoreBase::getNumberOfPlanes()
-{
-	return this->numberOfPlanes;
-}
-
-const std::vector<uint64_t>& BigCoreBase::getDataOrder()
-{
-    return dataOrder;
-}
-
-const std::vector<uint8_t>& BigCoreBase::getDataType()
-{
-    return dataType;
-}
-
-bool BigCoreBase::isEmpty()
-{
-	return this->dataLength == 0 ? true : false;
+    delete[] _data;
 }
 
 void BigCoreBase::clear()
 {
-	if (this->data) delete[] this->data;
+    numberOfImages = 0;
+    imageHeight = 0;
+    imageWidth = 0;
+    numberOfPlanes = 1;
+    dataOrder = { DataOrderIds::NUMBER_OF_IMAGES, DataOrderIds::IMAGE_HEIGHT, DataOrderIds::IMAGE_WIDTH, DataOrderIds::NUMBER_OF_PLANES };
+    dataType = { DataTypes::FLOAT };
+    delete[] _data;
+    _data = nullptr;
+    dataSize = 0;
+    memorySize = 0;
 }
 
-bool BigCoreBase::isInMemory()
+void BigCoreBase::setSupportingStructures()
 {
-	// TODO: implement
-	;
+    setDimensions();
+    setEntityTypeSizes();
+    setSubSizes();
+    setDataSize();
+    setOffsets();
+    setOrderMap();
 }
 
-void BigCoreBase::setMemorySize(uint64_t bytes)
+void BigCoreBase::setDimensions()
 {
-	this->memorySize = bytes;
+    dimensions.clear();
+    for (auto it = dataOrder.cbegin(); it != dataOrder.cend(); ++it) {
+        switch (*it) {
+        case DataOrderIds::NUMBER_OF_IMAGES:
+            dimensions.push_back(numberOfImages);
+            break;
+        case DataOrderIds::IMAGE_HEIGHT:
+            dimensions.push_back(imageHeight);
+            break;
+        case DataOrderIds::IMAGE_WIDTH:
+            dimensions.push_back(imageWidth);
+            break;
+        case DataOrderIds::NUMBER_OF_PLANES:
+            dimensions.push_back(numberOfPlanes);
+            break;
+        }
+    }
```

Comment:
Pri pouziti iteratoru bych radsi psal `for(auto it : dataOrder)` je to citelnejsi a preklada se to do totozneho byte kodu.